backend/src/GraphQl/Approvals/ApprovalMutations.cs

```diff
@@ -0,0 +1,116 @@
+        IDataService dataService,
+        IResponseApprovalService responseApprovalService,
+        CancellationToken cancellationToken
+    )
+    {
+        var currentUser = await userService.GetCurrentUser(
+            cancellationToken).ConfigureAwait(false);
+        if (currentUser is null)
+        {
+            return new AddApprovalPayload(
+                new AddApprovalError(
+                    AddApprovalErrorCode.UNAUTHENTICATED,
+                    $"The user is not authenticated.",
+                    []
+                )
+            );
+        }
+
+        if (!CommonAuthorization.IsAuthorizedToAddApprovalForInstitution(
+            currentUser,
+            input.CreatorId,
+            cancellationToken
+            )
+        )
+            return new AddApprovalPayload(
+                new AddApprovalError(
+                    AddApprovalErrorCode.UNAUTHORIZED,
+                    $"The current user is not authorized to approval for the institution.",
+                    []
+                )
+            );
+
+        var data = await dataService.GetDataAsync(input.DataId, context, cancellationToken).ConfigureAwait(false);
+        if (data == null)
+        {
+            return new AddApprovalPayload(
+                new AddApprovalError(
+                    AddApprovalErrorCode.UNKNOWN_DATA,
+                    $"Unknown data.",
+                    []
+                )
+            );
+        }
+
+        var approval = new DataApproval(
+                DateTime.Now,
+                input.Approval.Signature,
+                input.Approval.KeyFingerprint,
+                input.Approval.Query,
+                input.Approval.Response,
+                currentUser.Id);
+        if (input.Approval.Publication != null)
+        {
+            approval.Publication = new Publication(
+                input.Approval.Publication.Title,
+                input.Approval.Publication.Abstract,
+                input.Approval.Publication.Section,
+                input.Approval.Publication.Authors,
+                input.Approval.Publication.Doi,
+                input.Approval.Publication.ArXiv,
+                input.Approval.Publication.Urn,
+                input.Approval.Publication.WebAddress);
+        }
+        if (input.Approval.Standard != null)
+        {
+            approval.Standard = new Standard(
+                input.Approval.Standard.Title,
+                input.Approval.Standard.Abstract,
+                input.Approval.Standard.Section,
+                input.Approval.Standard.Year,
+                input.Approval.Standard.Standardizers,
+                input.Approval.Standard.Locator);
+        }
+        data.Approvals.Add(approval);
+        await context.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
+
+        try
+        {
+            ((Data.DataX)data).Approval = await responseApprovalService.CreateResponseApproval(data, cancellationToken).ConfigureAwait(false);
```

Comment:
Rather then casting, add the property `Approval` to the known type of the variable `data`. Every cast is a possible runtime error.